runtime_test.go

```diff
@@ -2163,3 +2251,195 @@ func BenchmarkAsciiStringMapGet(b *testing.B) {
 		}
 	}
 }
+
+func TestException(t *testing.T) {
+	const SCRIPT = `
+	function myFunc(times) {
+		if (times == 0) {
+			throw new Error("oh no!");
+		}
+		myFunc(times-1);
+	}
+	myFunc(5);`
+
+	r := New()
+	r.SetStackTraceLimit(3)
+	_, err := r.RunString(SCRIPT)
+
+	if err == nil {
+		t.Fatal("Expected err to not be nil")
+	}
+
+	ex, ok := err.(*Exception)
+	if !ok {
+		t.Fatalf("Expected err to be of type Exception, but got %T", err)
+	}
+
+	expected := `Error: oh no!
+	at myFunc (<eval>:4:10(8))
+	at myFunc (<eval>:6:9(13))
+	at myFunc (<eval>:6:9(13))
+`
+	if ex.String() != expected {
+		t.Fatalf("Expected: \n%s\n but got \n%s", expected, ex.String())
+	}
+}
+
+func TestExceptionWithinNativeFunction(t *testing.T) {
+	const SCRIPT = `
+	function myFunc() {
+		myNativeFunc();
+	}
+	myFunc(5);
+`
+
+	r := New()
+	native := func(call FunctionCall) Value {
+		if len(call.Arguments) < 1 {
+			panic(r.NewTypeError("oh no!"))
+		}
+
+		return UndefinedValue()
+	}
+	nativeFunc, nativeFuncErr := r.CreateNativeFunction("myNativeFunc", "", native)
+	if nativeFuncErr != nil {
+		t.Fatal("Expected nativeFuncErr to be nil")
+	}
+	if err := r.Set("myNativeFunc", nativeFunc); err != nil {
+		t.Fatal("Expected err to be nil")
+	}
+
+	_, err := r.RunString(SCRIPT)
+
+	if err == nil {
+		t.Fatal("Expected err to not be nil")
+	}
+
+	ex, ok := err.(*Exception)
+	if !ok {
+		t.Fatalf("Expected err to be of type Exception, but got %T", err)
+	}
+
+	// TODO: revisit this as it was the original expected value
```

Comment:
Calling out also a few tests here with a divergence from their original expected values.

Reply:
I assume this passes with the new expected stack trace? If so, this looks good to me